### DISCUSSION TITLE
Asymmetric cryptography

### DISCUSSION THREAD

**Discussion by JungeWerther** [supabase/supabase#29286] [Feature Requests] [answered]
As far as I understand, if I want to verify the validity of JWT claims, I need to validate their signature against the original signing secret. 

Now let's say I want to use different microservices to interact with my database on behalf of my users. The beauty of JWT is that, in principle, I wouldn't have to make a separate call to the the auth server for every back-end request, because I can validate tokens on-site.

However, this means I'd have to distribute the original signing secret across microservices, which means that if only one of them would get compromised, an attacker could impersonate the identity of all my users.

Is there some way to configure the use of asymmetric cryptographic algorithms for verifying the validity of supabase-issued JWTs, where if leaked, those keys could not be used to issue new JWTs?

**Comment by GaryAustin1** [comment]
https://github.com/supabase/auth-js/issues/873#issuecomment-2092401090
There is a plan.

**Comment by kangmingtay** [comment, marked as answer]
@JungeWerther it's on the way: https://github.com/orgs/supabase/discussions/29289

**Comment by JungeWerther** [comment]
Amazing !
